### ISSUE TITLE
Modify the serving port

### ISSUE THREAD

**Issue by Co1lin** [siyuan-note/siyuan#1990]
### In what scenarios do you need this function?

I want to run multiple backends of siyuan on a server to provide services for different users in my group. Then, they will have independent workspace. However, even with docker, I have failed doing so.

I found that although I mapped different ports such as `16806` to different siyuan containers, and specify corresponding addresses, all of the containers provided resources files through port 6806.

What's more, on some machines, 6806 may have been occupied.

### Describe the optimal solution

It seems easy to provide a command line parameter such as `--port=<my port>` to specify any port the user prefer. By the way, all of the paths of resources need to match the specified port.

**Comment by 88250** [comment]
Hello, please try to use a reverse proxy server (such as NGINX) to handle network traffic.

Here is a example, public port 80/443 proxy to host port 33733:

NGINX conf:

```nginx
location /udanax/s/1602224134353/ {
    proxy_pass http://127.0.0.1:33733/;
}
```

Docker port binding:

```go
PortBindings: nat.PortMap{
	"6806/tcp": []nat.PortBinding{
		{
			HostIP:   "127.0.0.1",
			HostPort: 33733,
		},
	},
},
```

I closed the issue. If you have any questions, please feel free to post.

**Comment by Co1lin** [comment]
Thanks for your reply.

Actually I have tried to use NGINX and docker port mapping, but I met some problems. I ran 
```shell
docker run --name syr -v ~/siyuan/conf:/siyuan/conf -v ~/siyuan/data:/siyuan/data -p 16806:6806 b3log/siyuan --resident=true --conf=/siyuan/conf/ --data=/siyuan/data/ --authCode=123 --readonly=true
```
to run the docker, and visit http://mydomain:16806 . However, the webpage stuck at the loading stage. I used Developer Tools to check what happened, and I saw 
```
Failed to load 'http://mydomain:6806/stage/vditor/dist/js/lute/lute.min.js?v=1.1.83'.
```
in the console, which means some resources were still provided with the fixed port 6806, which I think it was hard to solve.

By the way, the authentication requires both username and password. I can only find how to specify the password, but I don't know how to alter the username. The default username is "siyuan", which is figured out by my trying, and I suggest declare it in the user guide.
